lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -440,15 +440,7 @@ public Optional<TaskStatusProto> GetTaskStatus()
                 }
 
                 var taskStatusProto = _task.Value.GetStatusProto();
-                if (taskStatusProto.state == State.RUNNING)
-                {
-                    // only RUNNING status is allowed to rurn here, all other state pushed out to heartbeat 
-                    return Optional<TaskStatusProto>.Of(taskStatusProto);
-                }
-
-                var e = new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Task state must be RUNNING, but instead is in {0} state", taskStatusProto.state));
```

Comment:
I am not sure. 

From the comments of the caller, it says "the TaskStatusProto of the currently running task, if there is any", "the state of the running Task, if one is running." Meaning it expects the task is running. 

Do we ever see this exception ever happen? I didn't.